porcupine/plugins/highlight/tree-sitter-token-mappings/python.yml

```diff
@@ -202,3 +255,132 @@ token_mapping:
     type: Token.Name.Builtin
     vars: Token.Name.Builtin
     zip: Token.Name.Builtin
+
+    # builtin.Pseudo
+    self: Token.Name.Builtin.Pseudo
+    cls: Token.Name.Builtin.Pseudo
+    Ellipsis: Token.Name.Builtin.Pseudo
+    NotImplemented: Token.Name.Builtin.Pseudo
+
+    # Magic variables
+    __annotations__: Token.Name.Variable.Magic
+    __bases__: Token.Name.Variable.Magic
+    __class__: Token.Name.Variable.Magic
+    __closure__: Token.Name.Variable.Magic
+    __code__: Token.Name.Variable.Magic
+    __debug__: Token.Name.Variable.Magic
+    __defaults__: Token.Name.Variable.Magic
+    __dict__: Token.Name.Variable.Magic
+    __doc__: Token.Name.Variable.Magic
+    __file__: Token.Name.Variable.Magic
+    __func__: Token.Name.Variable.Magic
+    __globals__: Token.Name.Variable.Magic
+    __kwdefaults__: Token.Name.Variable.Magic
+    __loader__: Token.Name.Variable.Magic
+    __module__: Token.Name.Variable.Magic
+    __mro__: Token.Name.Variable.Magic
+    __name__: Token.Name.Variable.Magic
+    __objclass__: Token.Name.Variable.Magic
+    __package__: Token.Name.Variable.Magic
+    __path__: Token.Name.Variable.Magic
+    __qualname__: Token.Name.Variable.Magic
+    __self__: Token.Name.Variable.Magic
+    __slots__: Token.Name.Variable.Magic
+    __spec__: Token.Name.Variable.Magic
+    __weakref__: Token.Name.Variable.Magic
+
+    # Magic funcs
```

Comment:
I don't like the `__blah__` lines:
- They will easily get outdated as new things are added to Python, even more easily than other things in this file. I remember that in the past, pygments has not highlighted all of my magic variables I have used. I don't remember which magic variable that was though.
- They can confuse users: if you e.g. write `__init__` somewhere in the middle of a file, not in a class, you suddenly see it colored and think you're doing it right.
- When defining a magic method, `def __init__(...)` (which is by design more common than explicitly calling a magic method), you don't get the magic coloring anyway, as it is instead colored as a function name.

None of these are alone a big enough reason to remove the magic names, but it makes me dislike them overall. I'll let you decide though: if you want to keep them, we can keep them :)

Reply:
> They can confuse users: if you e.g. write `__init__` somewhere in the middle of a file, not in a class, you suddenly see it colored and think you're doing it right.

In fact Pygments does this too

> When defining a magic method, def __init__(...) (which is by design more common than explicitly calling a magic method), you don't get the magic coloring anyway, as it is instead colored as a function name.

Lol, my color scheme doesn't make difference between them, so I didn't even noticed this. I added these before the function definition checking logic, and therefore they worked.

Actually, now I don't like this either. It's almost 100 lines, and not even complete, although I copied them from Pygments